Create README.md from Wiki content (#342)
ros-perception/image_common: polled_camera/README.md

## `polled_camera` Overview

`polled_camera` defines the ROS interface that client nodes use to request images from a polling camera driver node (e.g. [prosilica_camera](https://index.ros.org/p/prosilica_camera/)). The protocol is:

> * The camera driver advertises a service call `<camera>/request_image` of type [polled_camera/GetPolledImage](http://www.ros.org/doc/api/polled_camera/html/srv/GetPolledImage.html).

Reviewer: link is for ROS 1, maybe use this one https://github.com/ros-perception/image_common/blob/rolling/polled_camera/srv/GetPolledImage.srv ?

Author: ```suggestion
* The camera driver advertises a service call `<camera>/request_image` of type [polled_camera/GetPolledImage](https://github.com/ros-perception/image_common/blob/rolling/polled_camera/srv/GetPolledImage.srv).
```

* The client calls the service, specifying an output namespace.
* On receiving a request, the driver captures an image and returns its time stamp in the service response.

> * The driver publishes (latching) the [sensor_msgs/Image](http://www.ros.org/doc/api/sensor_msgs/html/msg/Image.html) and [sensor_msgs/CameraInfo](http://www.ros.org/doc/api/sensor_msgs/html/msg/CameraInfo.html) to `<response_namespace>/image_raw` and `<response_namespace>/camera_info`.

Reviewer: Same here https://github.com/ros2/common_interfaces/blob/rolling/sensor_msgs/msg/Image.msg and https://github.com/ros2/common_interfaces/blob/rolling/sensor_msgs/msg/CameraInfo.msg

You might use this other one https://docs.ros2.org/foxy/api/sensor_msgs/msg/Image.html ? but `rolling` is not working.

Author: Yeah we need to get better canonical links for messages.

* Clients subscribe to the response topics just like any other camera image stream.
* When a disconnection causes the number of subscribers to drop to zero, the publisher shuts down.

See the [code API](http://www.ros.org/doc/api/polled_camera/html/) (unstable) for more information on writing polled camera drivers or clients.

There are several reasons for publishing images to topics instead of returning them from the service call:

* **Consistency:** Client-side processing is always performed in an image callback, regardless of what type of camera is used.

> * **Visibility:** Topics can be monitored in [image_view](https://index.ros.org/p/image_view/) or [rviz](https://index.ros.org/p/rviz/).

Reviewer: - rviz https://github.com/ros2/rviz
 - https://docs.ros.org/en/rolling/p/image_view/doc/index.html

Author: Same canonical link preference.

> * **Reproducibility:** Topics can be [bagged](https://index.ros.org/p/rosbag/) and played back later.

Reviewer: https://github.com/ros2/rosbag2?tab=readme-ov-file#using-rosbag2

Author: ```suggestion
* **Reproducibility:** Topics can be [bagged](https://index.ros.org/p/rosbag2/) and played back later.
```

Same comment here about using the canonical link

> * **Compression:** Can take advantage of [image_transport](https://index.ros.org/p/image_transport/) to compress images, if desired.

Reviewer: https://docs.ros.org/en/rolling/p/image_transport/

Author: I'd like to work to get the canonical link for packages to be the index page which is much better cross linked. It has a link directly to the API docs for each different version.  We do need to get more README's ported. I'm doing this as a test of migrating content from the wiki . If we migrated to it's wiki page ( https://wiki.ros.org/image_transport) to it's README.md the link here would be much more useful. But as we're looking to wind down the wiki, we are considering a more bulk process and I'd like to do the linking with that in mind.

## Nodes

### poller

Continually requests images from a polled camera at a specified rate in Hz, making the camera look like it is capturing continuously.

Usage:

```bash
# Poll "my_camera" at 5 Hz, publishing in namespace my_polled_output/.

> $ poller 5 camera:=my_camera output:=my_polled_output

Reviewer: is this correct ?

```bash
ros2 run polled_camera poller 5 camera:=my_camera output:=my_polled_output
```

Author: I don't have a clue, I'm just porting the documentation as it exists trying to apply some basic heuristics.

```

Published Topics
* <output>/image_raw (sensor_msgs/Image)
  Image topic, actually published by the camera driver. output should be remapped by the user.
* <output>/camera_info (sensor_msgs/CameraInfo)
  Info topic, actually published by the camera driver. output should be remapped by the user.
Services Called
* <camera>/request_image (polled_camera/GetPolledImage)

> The camera driver's polled image service. camera should be remapped by the user to the camera namespace.

Reviewer: ```suggestion
Published Topics
* `<output>/image_raw` ([sensor_msgs/Image](https://github.com/ros2/common_interfaces/blob/rolling/sensor_msgs/msg/Image.msg))
  Image topic, actually published by the camera driver. output should be remapped by the user.
* `<output>/camera_info` ([sensor_msgs/CameraInfo](https://github.com/ros2/common_interfaces/blob/rolling/sensor_msgs/msg/CameraInfo.msg))
  Info topic, actually published by the camera driver. output should be remapped by the user.
Services Called
* `<camera>/request_image` ([polled_camera/GetPolledImage](https://github.com/ros-perception/image_common/blob/rolling/polled_camera/srv/GetPolledImage.srv))
The camera driver's polled image service. camera should be remapped by the user to the camera namespace.
```
